### PR TITLE
Cache the artifact nexus data bag

### PR DESCRIPTION
**Pull request by KAllan357**
All those GETs are probably way more expensive than they seem for a long Chef run.

### CODE REVIEW

**Comment by KAllan357** [comment]
Also attempt to fix https://github.com/RiotGames/artifact-cookbook/issues/67

**Comment by jhowarth** [comment]
Can we do this and still perserve backwards compatibility?  Maybe add a deprecation warning.

File: libraries/chef_artifact.rb [hunk 1]
```diff
@@ -55,29 +55,19 @@ def windows?
...
+      # 
+      # @return [Chef::DataBagItem] the data bag item
       def nexus_config_for(node)
-        data_bag_item = load_nexus_dbi
-
-        config = data_bag_item[node.chef_environment] || data_bag_item["*"]
-        unless config
-          raise EnvironmentNotFound.new(NEXUS_DBI, node.chef_environment)
+        data_bag_item = if Chef::Config[:solo]
+          Chef::DataBagItem.load(DATA_BAG, WILDCARD_DATABAG_ITEM)
```
Comment (jhowarth) [inline libraries/chef_artifact.rb:66]: This part isn't backwards compatible, should it be?

**Comment by jhowarth** [comment]
:+1: after that last comment.

**Comment by ivey** [comment]
Seems good to me.

**Pull request merged**